voltdbclient/query_api.go

```diff
@@ -148,7 +148,7 @@ func (c *Conn) QueryTimeout(query string, args []driver.Value, timeout time.Dura
 		case resp := <-pi.responseCh:
 			switch e := resp.(type) {
 			case VoltRows:
-				return e, nil
+				return &e, nil
```

Comment:
Can you explain why you need to change the return type to pointer here? 
If so, function return types also need update.

Reply:
Because I needed to add `func (vr *VoltRows) NextResultSet() error` which needs to change the current table index. Function return type does not need an update as it is an interface itself.

Reply:
I see. But I assume the the query result would be stored to some local variable before issue the NextResultSet() method, right? So that nextResultSet still be able to call upon that variable? i.e 
```
rows, err := conn.Query(....)
for rows.HasNextResultSet() {
      .....
      rows.NextResultSet()  // x.m() is shorthand for (&x).m() ? https://golang.org/ref/spec#Calls
}
```

Reply:
So yes Go has automatic dereferencing of pointers but that's not the issue here. 

The problem is that because we must update `vr.tableIndex` I must use a pointer receiver in the `NextResultSet` method  (`func (vr *VoltRows) NextResultSet()`). If purely working on the VoltRows object itself this is no issue. However VoltRows needs to implement driver.RowsNextResultSet as this is the enhancement interface `sql/database` tests against for 
this logic. Here lies the problem, if providing the composite literal `VoltRows{}` instead of a pointer to it `&VoltRows{}`; it will not satisfy the interface and `sql/database` will assume the driver does not support multiple result sets and NextResultSet will always return false.

I've added a unit test so this can easily be demonstrated as well as a compile time assertion for the `driver.RowsNextResultSet` interface. Remove the `&` and see stuff fail.

Reply:
I see, that make sense. Thanks for the detailed explanation.